### PR TITLE
Rework parameters

### PR DESCRIPTION
**Pull request by KaQuMiQ**

### CODE REVIEW

File: src/draive/types/model.py [hunk 1]
```diff
@@ -1,35 +1,16 @@
...
-                    for parameter in cls._parameters()
-                    if parameter.default is MISSING
-                ],
-            }
-
-        return cls._specification
-
-    @classmethod
+class Model(ParametrizedModel):
+    @classmethod  # avoid making json each time
```
Comment (piotr-rarus) [inline src/draive/types/model.py:13]: For immutable objects you can add `@cached_property` decorator so they are only calculated once.

File: tests/test_model.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        "value_default": 8,
+        "invalid": "valid",
+        "answer": {"string": "value", "more": None},
+        "all": True,
+    }
+    assert json.loads(ExampleModel.validated(**values_dict).as_json()) == values_dict
+
+
+def test_validated_passes_with_default_values() -> None:
+    ExampleModel.validated()
```
Comment (piotr-rarus) [inline tests/test_model.py:47]: What does `validated` method returns? Can we put some assert here?

Comment (KaQuMiQ) [reply tests/test_model.py:47]: those are temporary tests, we have to make a proper test suite here - here it should be just made with defaults, we could make some asserts but as I said - we will make proper tests here soon

File: src/draive/types/model.py (hunk 1 above)
Comment (KaQuMiQ) [reply src/draive/types/model.py:13]: it might not work properly with classmethods 🤔

Comment (piotr-rarus) [reply src/draive/types/model.py:13]: It was just a general comment.

**Review by piotr-rarus — APPROVED** [review]

**Pull request merged**